src/graphics/Window.cpp

```diff
@@ -0,0 +1,85 @@
+//
+// Created by ajer on 11.02.19.
+//
+
+#include "Window.h"
+
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+
+namespace Bald::Graphics {
+
+    Window::Window(GLFWwindow* glfwwindow, const char* title, int width, int height, bool VSync):
+        m_Window(glfwwindow),
+        m_Title(title),
+        m_Width(width),
+        m_Height(height),
+        m_VSync(VSync)
+    {
+        Window::Init();
+    }
+
+    Window::~Window() noexcept {
+        Shutdown();
+    }
+
+    void Window::Update() noexcept {
+        glfwPollEvents();
+        glfwSwapBuffers(m_Window);
+    }
+
+    /** TODO: Callbacks
+     * glfwGetFramebufferSize(m_Window, &m_Width, &m_Height);
+     * glViewport(0, 0, m_Width, m_Height);
+     */
+
+    void Window::WindowSetVSync(bool enabled) noexcept {
+        if (enabled)
+            glfwSwapInterval(1);
+        else
+            glfwSwapInterval(0);
+
+        m_VSync = enabled;
+    }
+
+
+    bool Window::Init() noexcept {
+
+
+        if (!gladLoadGLLoader(reinterpret_cast<GLADloadproc>(glfwGetProcAddress))) {
+            CORE_LOG_ERROR("[Window] Failed to load GLAD!");
+            exit(1);
+        }
+
+        if (!glfwInit()) {
+            CORE_LOG_ERROR("[Window] Failed to init GLFW!");
+            exit(1);
+        }
+
```

Comment:
after initializing all OpenGL we should print all the information about it and about computer we currently running

Reply:
!!!